crates/walrus-service/src/node/storage.rs

```diff
@@ -105,8 +113,17 @@ pub enum NodeStatus {
     RecoverMetadata,
     /// The node is in recovery mode and catching up with the chain.
     RecoveryCatchUp,
-    /// The node is in recovery mode and recovering missing slivers.
+    /// The node is in recovery mode and recovering missing slivers. The included epoch is the
+    /// epoch in which the node started recovering.
     RecoveryInProgress(Epoch),
+    /// The node is in recovery mode and catching up with the chain, but the history is incomplete
+    /// due to expired event blobs.
+    RecoveryCatchUpWithIncompleteHistory {
+        /// The first epoch for which all events are available and relevant.
+        first_complete_epoch: Epoch,
+        /// The epoch at which the node started recovering.
+        epoch_at_start: Epoch,
```

Comment:
Nit: can we document the function of these two fields so that it's clear why we include them?

IIUC: first_complete_epoch is used to filter events that before this epoch, and epoch_at_start is for processing events that have no effect up to this epoch?